.travis.yml

```diff
@@ -210,14 +212,5 @@ jobs:
         - sudo cp odo /usr/bin
         - export KUBERNETES=true
         - travis_wait make test-cmd-project
-        - travis_wait make test-cmd-devfile-catalog
-        - travis_wait make test-cmd-devfile-watch
-        - travis_wait make test-cmd-devfile-push
-        - travis_wait make test-cmd-devfile-debug
-        - travis_wait make test-cmd-devfile-exec
-        - travis_wait make test-cmd-devfile-delete
-        - travis_wait make test-cmd-devfile-create
-        - travis_wait make test-cmd-devfile-url
-        - travis_wait make test-cmd-devfile-test
-        - travis_wait make test-cmd-devfile-log
-        - travis_wait make test-cmd-devfile-storage
+        - travis_wait make test-integration-devfile
```

Comment:
Sometimes we don't want to run a particular job on kubernetes however we do run for openshift. For that type of scenario it won't be good to run the job as a whole in one go.

Reply:
> Sometimes we don't want to run a particular job on kubernetes

We support all the devfile file test run on Kubernetes.

Reply:
Yes true currently we support it but what about the time when we were only running `test-cmd-devfile-url` on openshift. If that kind of scenario comes again we need to jump on current format.

Reply:
For 2.0, devfile changes are no more an experimental feature and whatever the issue we had that all has been fixed. It is just to make sure that for any new up coming devfile changes should support Kubernetes cluster too. The benefit of calling the whole target is contributor does not bother each time to add new target in CI file when a new test file is added within devfile test package.

Reply:
> The benefit of calling the whole target is contributor does not bother each time to add new target in CI file when a new test file is added within devfile test package.

Indeed, But devfile is not all about 2.0 only. We still have some of the unknown feature which are not considered for 2.0 . Although I am not sure how efficient it will be in identifying failures also. TBH I am also concerned about the logs which we will get only after the complete execution of `make test-integration-devfile`. However if it would have run individually and the particular jobs stuck in between we can get the failure and fix it much quicker.

Reply:
This is what I was talking about

```
$ travis_wait make test-integration-devfile
Still running (20 of 20): make test-integration-devfile
Timeout (20 minutes) reached. Terminating "make test-integration-devfile"
The command make test-integration-devfile exited with 137.
Log:
ginkgo  -randomizeAllSpecs -slowSpecThreshold=120 -timeout 7200s -nodes=2 tests/integration/devfile/
Running Suite: Devfile Suite
============================
Random Seed: 1598268410 - Will randomize all specs
Will run 169 specs
Running in parallel across 2 nodes
•••••••••••
------------------------------
S [SKIPPING] in Spec Setup (JustBeforeEach) [5.541 seconds]
odo devfile url command tests
/home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_url_test.go:16
  Testing URLs for OpenShift specific scenarios
  /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_url_test.go:268
    should list URLs for s2i components [JustBeforeEach]
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_url_test.go:399
    This is a OpenShift specific scenario, skipping
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_url_test.go:271
------------------------------
•••••
------------------------------
S [SKIPPING] in Spec Setup (JustBeforeEach) [5.504 seconds]
odo devfile create command tests
/home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:15
  When executing odo create with component type argument and --s2i flag
  /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:132
    should fail the create command as --git flag, which is specific to s2i component creation, is used without --s2i flag [JustBeforeEach]
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:171
    Skipping test because s2i image is not supported on Kubernetes cluster
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:136
------------------------------
••••••••
------------------------------
S [SKIPPING] in Spec Setup (JustBeforeEach) [5.601 seconds]
odo devfile create command tests
/home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:15
  When executing odo create with component type argument and --s2i flag
  /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:132
    should fail to create the devfile component which doesn't have an s2i component of same name [JustBeforeEach]
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:167
    Skipping test because s2i image is not supported on Kubernetes cluster
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:136
------------------------------
•••
------------------------------
S [SKIPPING] in Spec Setup (JustBeforeEach) [5.493 seconds]
odo devfile create command tests
/home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:15
  When executing odo create with component type argument and --s2i flag
  /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:132
    should fail the create command as --now flag, which is specific to s2i component creation, is used without --s2i flag [JustBeforeEach]
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:183
    Skipping test because s2i image is not supported on Kubernetes cluster
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:136
------------------------------
••••
------------------------------
S [SKIPPING] in Spec Setup (JustBeforeEach) [5.849 seconds]
odo devfile create command tests
/home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:15
  Disabling experimental preference should show a disclaimer
  /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:58
    checks that the experimental warning does *not* appear when Experimental is set to false for create [JustBeforeEach]
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:64
    Skipping test because s2i image is not supported on Kubernetes cluster
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:61
------------------------------
•••••••••••••••••
------------------------------
S [SKIPPING] [5.619 seconds]
odo devfile push command tests
/home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_push_test.go:21
  push with listing the devfile component
  /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_push_test.go:675
    checks devfile and s2i components together [It]
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_push_test.go:719
    Skipping test because s2i image is not supported on Kubernetes cluster
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_push_test.go:721
------------------------------
•••••••••••••••••••••••••••••••
------------------------------
S [SKIPPING] in Spec Setup (JustBeforeEach) [5.526 seconds]
odo devfile create command tests
/home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:15
  When executing odo create with component type argument and --s2i flag
  /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:132
    should fail the create command as --binary flag, which is specific to s2i component creation, is used without --s2i flag [JustBeforeEach]
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:176
    Skipping test because s2i image is not supported on Kubernetes cluster
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:136
------------------------------
••••••
------------------------------
S [SKIPPING] in Spec Setup (JustBeforeEach) [5.491 seconds]
odo devfile create command tests
/home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:15
  When executing odo create with component type argument and --s2i flag
  /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:132
    should successfully create the localconfig component with --git flag [JustBeforeEach]
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:156
    Skipping test because s2i image is not supported on Kubernetes cluster
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_create_test.go:136
------------------------------
••••••••••
------------------------------
S [SKIPPING] in Spec Setup (JustBeforeEach) [5.880 seconds]
odo devfile url command tests
/home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_url_test.go:16
  Testing URLs for OpenShift specific scenarios
  /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_url_test.go:268
    should create a url for a unsupported devfile component [JustBeforeEach]
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_url_test.go:382
    This is a OpenShift specific scenario, skipping
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_url_test.go:271
------------------------------
S [SKIPPING] in Spec Setup (JustBeforeEach) [5.814 seconds]
odo devfile url command tests
/home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_url_test.go:16
  Testing URLs for OpenShift specific scenarios
  /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_url_test.go:268
    should describe appropriate Route URLs [JustBeforeEach]
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_url_test.go:356
    This is a OpenShift specific scenario, skipping
    /home/travis/gopath/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_url_test.go:271
------------------------------
••••••The command "travis_wait make test-integration-devfile" exited with 137.
```
We can not see where the failure is. Which job ran into failure. I would prefer running individual make target on travis CI.

Reply:
> We still have some of the unknown feature which are not considered for 2.0 

Can you please specify those ? i mean devfile feature run on ocp but should not run on kubernetes cluster.

> Although I am not sure how efficient it will be in identifying failures also. TBH I am also concerned about the logs which we will get only after the complete execution of make test-integration-devfile. However if it would have run individually and the particular jobs stuck in between we can get the failure and fix it much quicker.

You will get the same log as it was on failure spec. In best case scenario It definitely helps to get the failure result quicker as it does live log streaming in the job console, for example if you hit the issue in the first test target ```make test-cmd-devfile-catalog```. When you have multiple combination of test platform the demerit of calling each target will unnecessarily increase the maintains effort, for example each time you need to add the new target to the travis file. And the worst part is till now we do not know that we have never run devfile env and devfile registry test on kubernetes cluster since long time. 

> We can not see where the failure is. Which job ran into failure. I would prefer running individual make target on travis CI.

Timeout (20 minutes) reached. Terminating "make test-integration-devfile"
The command make test-integration-devfile exited with 137.

The default job timeout (different from ginkgo test node timeout) for a travis job is 20 min causing the issue. We can overwrite that value.